documentation/CAMARA-Security-Interoperability.md

```diff
@@ -0,0 +1,231 @@
+  *  Cross-Site Request Forgery Protection
+* Client-Initiated Backchannel Authentication Flow
+  *  Authentication Request
+* Client Credentials Flow
+* Handling of acr_values
+* Access Token Request
+* The Scope Parameter
+* Missing "openid" scope
+* Purpose
+* ID Token sub claim
+* Client Authentication
+* OpenId Foundation Certification
+* References
+<!-- TOC -->
+
+
+## Introduction
+
+This document is the CAMARA Security and Interoperability Profile. To ensure interoperability and increased security, this technical specification restricts some options available in OIDC and CIBA, but does not change these standards.
+
+The CAMARA document sharpens the following for interoperability and security: 
+
+* Of the flows defined in OIDC CAMARA uses the [OIDC Authorization Code Flow](https://openid.net/specs/openid-connect-core-1_0.html#CodeFlowAuth)
+
+* Scope Parameter. Recommendations about the format of the scope parameter used in  CAMARA APIs.
+
+* Client Authentication. Specifications for client authentication within CAMARA.
+
+By encapsulating these elements within this document, it aims to provide a comprehensive guide for developers and operators, ensuring consistent implementation and adherence to standardized security measures across the CAMARA ecosystem. The defined OIDC profile not only facilitates the integration process, but also serves as a basic framework for developers wishing to leverage the CAMARA APIs while maintaining security and interoperability.
+
+
+## Audience
+
+The target audience for this document is the service/technical departments of operators exposing network functions via standard CAMARA APIs. And applications or client systems that consume CAMARA standard APIs to make use of the operator's network capabilities.
+
+
+## Conventions
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+## General Considerations
+
+### Transport Security
+All network connections MUST use TLS 1.2 or better.
+
+## OIDC Authorization Code Flow
+
+The OIDC Authorization Code Flow is defined in [OpenID Connect](https://openid.net/specs/openid-connect-core-1_0.html)
+
+### Cross-Site Request Forgery Protection
+
+CAMARA REQUIRES cross-site request forgery protection.
+
+CAMARA RECOMMENDS PKCE for CSRF protection. 
+CAMARA Authorization Servers SHOULD implement PKCE. If PKCE is not used by the Client then the CAMARA AZ must handle **state** and **nonce** as defined in OAuth2.
+
+CAMARA Clients SHOULD use PKCE if the CAMARA AZ supports PKCE.
+
+If nonce for CSRF-protection is used then implementers must ensure that sufficient entropy is present in the nonce value.
+Please see [OAuth 2.0 Security Best Current Practice](https://oauthstuff.github.io/draft-ietf-oauth-security-topics/draft-ietf-oauth-security-topics.html#name-protecting-redirect-based-f).
+
+### Offline Access
```

Comment:
This section should be revisited with whatever we end up agreeing on in the PR #123 discussion (see https://github.com/camaraproject/IdentityAndConsentManagement/pull/123#issuecomment-1966319812).

Reply:
@jpengar I just proposed/asked in a [review comment ](https://github.com/camaraproject/IdentityAndConsentManagement/pull/123#discussion_r1504134748)to close PR https://github.com/camaraproject/IdentityAndConsentManagement/pull/123 and have the #121 text here and not in documentation/CAMARA-API-access-and-user-consent.md

Reply:
> @jpengar I just proposed/asked in a https://github.com/camaraproject/IdentityAndConsentManagement/pull/123#discussion_r1504134748to close PR https://github.com/camaraproject/IdentityAndConsentManagement/pull/123 and have the https://github.com/camaraproject/IdentityAndConsentManagement/pull/121 text here and not in documentation/CAMARA-API-access-and-user-consent.md

We can discuss this in PR https://github.com/camaraproject/IdentityAndConsentManagement/pull/123. But first, please see my comments in https://github.com/camaraproject/IdentityAndConsentManagement/pull/123#issuecomment-1966319812 regarding the proposed text content to be added to the OIDC profile regarding offline access. I don't agree with all of the proposed text.